Resource deletion and re-deployment causes app crash

```
panic: interface conversion: interface {} is *v1.NexusAlgorithmWorkgroup, not *v1.NexusAlgorithmTemplate [recovered]

	panic: interface conversion: interface {} is *v1.NexusAlgorithmWorkgroup, not *v1.NexusAlgorithmTemplate


goroutine 130 [running]:

k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x254fa30, 0x4000cb5e60}, {0x1d96020, 0x4000cb5e30}, {0x0, 0x0, 0x4000cebc58?})

	/app/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:114 +0x16c

k8s.io/apimachinery/pkg/util/runtime.HandleCrashWithLogger({{0x2556160?, 0x4000686390?}, 0x4000cebca8?}, {0x0, 0x0, 0x0})

	/app/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:91 +0x104

panic({0x1d96020?, 0x4000cb5e30?})

	/usr/local/go/src/runtime/panic.go:792 +0x124

github.com/SneaksAndData/nexus/services.(*NexusResourceCache).onConfigurationDeleted(0x4000cebe58?, {0x2101540, 0x40009bc900})

	/app/services/configuration_cache.go:101 +0x118

k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(...)

	/app/vendor/k8s.io/client-go/tools/cache/controller.go:271

k8s.io/client-go/tools/cache.(*processorListener).run.func1(0x4000cebf4f, 0x40006e00a0, {0x1e19ce0?, 0x4000d2f290?})

	/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:1082 +0x144

k8s.io/client-go/tools/cache.(*processorListener).run(0x40006e00a0)

	/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:1087 +0x3c

k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()

	/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:72 +0x58

created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start in goroutine 103

	/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:70 +0x7c
```

--- a/services/configuration_cache.go
+++ b/services/configuration_cache.go
@@ -78,7 +78,7 @@ func (c *NexusResourceCache) onConfigurationAdded(obj interface{}) {
 		return
 	}
 
-	c.logger.V(3).Info("resource loaded", "algorithm", objectRef.Name)
+	c.logger.V(3).Info("resource loaded", "resource", objectRef.Name)
 }
 
 func (c *NexusResourceCache) onConfigurationUpdated(old, new interface{}) {
@@ -98,7 +98,13 @@ func (c *NexusResourceCache) onConfigurationUpdated(old, new interface{}) {
 	c.logger.V(3).Info("resource updated", "resource", newRef.Name, "diff", diff.ObjectGoPrintSideBySide(old, new))
 }
 func (c *NexusResourceCache) onConfigurationDeleted(obj interface{}) {
-	c.logger.V(3).Info("resource deleted", "resource", obj.(*v1.NexusAlgorithmTemplate).Name)
+	objectRef, err := cache.ObjectToName(obj)
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+
+	c.logger.V(3).Info("resource deleted", "resource", objectRef.Name)
 }
 
 // GetAlgorithmConfiguration retrieves a cached NexusAlgorithmTemplate resource from informer cache

